tests/testthat/test_Plot_synergy.R

```diff
@@ -0,0 +1,214 @@
+test_that("color_ranges_from_green_to_white_to_red", {
+  expected_colors <-
+    c("#00FF00", "#3FFF3F", "#7FFF7F", "#BFFFBF", # 00ff00 is green
+      "#FFFFFF", "#FFBFBF", "#FF7F7F", "#FF3F3F", # ffffff is white
+      "#FF0000")                                  # ff0000 is red
+  expect_equal(expected_colors, .GetColors(-8,8))
+})
+
+test_that("slice_at_least_first_works", {
+  expect_equal(2:5, .SliceAtLeastFirst(c(-1,5)))  # pathological case
+  expect_equal(2:5, .SliceAtLeastFirst(c(0,5)))   # degenerate case
+  expect_equal(2:5, .SliceAtLeastFirst(c(1,5)))
+  expect_equal(2:5, .SliceAtLeastFirst(c(2,5)))
+  expect_equal(3:5, .SliceAtLeastFirst(c(3,5)))
+  expect_equal(3:5, .SliceAtLeastFirst(c(3,5,10)))  # spurious val 10
+})
+
+test_that("numeric_apply_range_works", {
+  val1 <- c("1", 2, 3)
+  val2 <- c('2', 4, '6', 8, '10')
+  expect_equal(c(1,2,3), .NumericApplyRange(val1, NULL))
+  expect_equal(c(1,2,3), .NumericApplyRange(val1, c(1,3)))
+  expect_equal(c(1,2,3), .NumericApplyRange(val1, c(1,3,5))) # spurious val 5
+  expect_equal(c(2,4,6,8,10), .NumericApplyRange(val2, NULL))
+  expect_equal(c(4,6,8), .NumericApplyRange(val2, c(2,4)))
+})
+
+test_that("apply_ranges_works", {
+  data <- matrix(c(1,2,3,
+                   4,5,6,
+                   7,8,9), nrow = 3, ncol = 3, byrow=TRUE)
+
+  # Various ways to apply ranges that change nothing.
+  expect_equal(data, .ApplyRanges(data, NULL, NULL))    # no ranges
+  expect_equal(data, .ApplyRanges(data, c(1,3), NULL))  # full row range
+  expect_equal(data, .ApplyRanges(data, NULL, c(1,3)))  # full col range
+  expect_equal(data, .ApplyRanges(data, c(1,3), c(1,3))) # both full ranges
```

Comment:
1. .ApplyRanges also accepts spurious cases? e.g. row.range = c(1, 3, 5).
2. Exceed the full ranges cases: expect_equal(data, .ApplyRanges(data, c(1, 5), c(1, 5))?

Reply:
1. Done.
2. This is an error.  If I were going full-professional here I would design a section of this unit test to expect an R exception for this case, and verify that I get the exception -- but if it's all right with you I'd like to leave that as an exercise for the maintainer. :)  Meanwhile, I updated the comment for .ApplyRanges() to note that asking for the first 5 elements of a 3x3 matrix will yield an error.

A more interesting question is, *should* .ApplyRanges() automatically translate 5->3 for that 3x3 matrix.  IMHO it's better not to silently do that, because otherwise someone might believe they had a 7x7 matrix and got the first 5 elements, when really they had a 3x3 and got the first 3.  It could make for hard/confusing debugging several steps later.